src/tools/actors/ActorExecutionTrace.java

```diff
@@ -1,84 +1,401 @@
+          b.putShort((short) idx);
+        }
+
+        if (em instanceof PromiseSendMessage) {
+          b.put(Events.PromiseMessage.id);
+          b.putLong(((PromiseMessage) em).getPromise().getPromiseId());
+        } else {
+          b.put(Events.BasicMessage.id);
+        }
+
+        b.putLong(em.getSender().getActorId()); // sender
+        b.putLong(em.getCausalMessageId());
+        b.putShort(em.getSelector().getSymbolId());
+        b.putLong(m.getMessageExecutionStart(idx));
+        b.putLong(m.getMessageSendTime(idx));
+
+        Object[] args = em.getArgs();
+        b.put((byte) (args.length - 1)); //num paramaters
+
+        for (int i = 1; i < args.length; i++) {
+          //gonna need a 8 plus 1 byte for most parameter, boolean just use two identifiers.
+          if (args[i] instanceof SFarReference) {
+            Object o = ((SFarReference) args[i]).getValue();
+            writeParameter(o, b);
+          } else {
+            writeParameter(args[i], b);
+          }
+        }
+
+        idx++;
+      }
+    }
+
+    logMemoryUsage();
+  }
+
+  private static void writeParameter(final Object param, final ByteBuffer b) {
+    if (param instanceof SPromise) {
+      b.put((byte) 0x04);
+      b.putLong(((SPromise) param).getPromiseId());
+    } else if (param instanceof SResolver) {
+      b.put((byte) 0x05);
+      b.putLong(((SResolver) param).getPromise().getPromiseId());
+    } else if (param instanceof SAbstractObject) {
+      b.put((byte) 0x06);
+      b.putShort(((SAbstractObject) param).getSOMClass().getName().getSymbolId());
     } else {
-      processedMessages = null;
+      if (param instanceof Long) {
+        b.put((byte) 0x02);
+        b.putLong((Long) param);
+      } else if (param instanceof Double) {
+        b.put((byte) 0x03);
+        b.putDouble((Double) param);
+      } else if (param instanceof Boolean) {
+        if ((Boolean) param) {
+          b.put((byte) 0x01);
+        } else {
+          b.put((byte) 0x00);
+        }
+      } else if (param instanceof String) {
+        b.put((byte) 0x07);
+      } else {
+
+        throw new RuntimeException("unexpected parameter type");
+      }
     }
-    return processedMessages;
   }
 
-  public static void clearProcessedMessages() {
-    for (ObjectBuffer<ObjectBuffer<EventualMessage>> o : messagesProcessedPerThread) {
-      o.clear();
+  /**
+   *
+   * @param fc The FrontendConnector used to send data to the debugger.
+   */
+  public static void setFrontEnd(final FrontendConnector fc) {
+    front = fc;
+  }
+
+  public static synchronized void logSymbol(final SSymbol symbol) {
+    symbolsToWrite.add(symbol);
+  }
+
+  private static class TraceWorkerThread extends Thread{
+    @Override
+    public void run() {
+      File folder = new File(System.getProperty("user.dir") + "/traces");
+      folder.mkdir();
+
+      File f = new File(folder.getAbsolutePath() + "/" + new Date().toString() + ".trace");
```

Comment:
would be good if the trace file names could be configured as a option via the `./som` launcher, possibly passed in via some `-Dsom.trace$$` java property.
Then the trace file has a deterministic name and can be handled easier from scripts, which will be important for experiments.